apps/docker-base/Makefile.docker-base

```diff
@@ -1,35 +1,40 @@
-# -*- makefile -*-
+# Makefile for Docker image setup
+
 help:
 	@echo "This should be run as part of the Dockerfile"
 	false
 
 user-setup:
+	# Create app group and user
 	groupadd --gid 1000 app
 	useradd -d /app --uid 1000 --gid app app
 	chown -R app:app /app
 
 apt-setup:
+	# Configure APT to keep downloaded packages
 	rm -f /etc/apt/apt.conf.d/docker-clean
 	echo 'Binary::apt::APT::Keep-Downloaded-Packages "true";' > /etc/apt/apt.conf.d/keep-cache
 
 apt-install:
-	DEBIAN_FRONTEND=noninteractive apt update
-	# gcc and python3-dev needed on arm for guidance
-	DEBIAN_FRONTEND=noninteractive apt -y install --no-install-recommends python3-poetry gcc python3-dev
+	# Install necessary packages without recommended packages
+	DEBIAN_FRONTEND=noninteractive apt update && \
+	DEBIAN_FRONTEND=noninteractive apt -y install --no-install-recommends python3-poetry gcc python3-dev && \
+	# Clean up to reduce image size
+	apt clean && rm -rf /var/lib/apt/lists/*
 
 non-root-files-check:
+	# Check for any files owned by root
 	find . -uid 0 -ls
 	test $$(find . -uid 0 -print | wc -w) = 0
 
 record-version:
+	# Ensure GIT_COMMIT is set and record the version
 	test "$(GIT_COMMIT)" != ""
 	test "$(GIT_COMMIT)" != "unknown"
 	touch .git.commit.$(GIT_COMMIT)
 
-# Allow images that depend on the docker base image to verify that the version for their
```

Comment:
Please don't get rid of the longer comment that explains why we are doing this.
Feel free to move it under the target for consistency, but readers of the code need to understand what version consistency is being checked and why.